k8sutil/watch_test.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from .watch import Watch
+from mock import Mock
```

Comment:
Thanks for the video. It was useful. I agree that complicated Mocks will make things harder to follow and keeping test/mock with code would be harder. My point here is the Mock is a simple request mock and does not have much logic. Maybe if I refactor the test itself, it would be more clear. Nevertheless, you refactoring suggestions are good and I am going to apply them as much as possible.